synapse/util/caches/descriptors.py

```diff
@@ -232,7 +234,7 @@ def __get__(self, obj: Optional[Any], owner: Optional[Type]) -> Callable[..., An
         get_cache_key = self.cache_key_builder
 
         @functools.wraps(self.orig)
-        def _wrapped(*args: Any, **kwargs: Any) -> Any:
+        def _wrapped(*args: Any, **kwargs: Any) -> "defer.Deferred[Any]":
```

Comment:
Aside: there might be scope to give types to `self.orig`'s arguments with a `ParamSpec` here in the future.

(In fact, I wonder if Python typing is expressive enough these days to not need a mypy plugin for this at all?)

Reply:
> (In fact, I wonder if Python typing is expressive enough these days to not need a mypy plugin for this at all?)

I also had this thought but didn't want to go down that avenue at the moment.